src/ripple/core/impl/Config.cpp

```diff
@@ -358,274 +359,268 @@ void Config::setup (std::string const& strConf, bool bQuiet)
-                    FETCH_DEPTH = 1000000000u;
-                else
-                    FETCH_DEPTH = beast::lexicalCastThrow <std::uint32_t> (strTemp);
+    if (getSingleSection (secConfig, SECTION_FETCH_DEPTH, strTemp))
+    {
+        boost::to_lower (strTemp);
 
-                if (FETCH_DEPTH < 10)
-                    FETCH_DEPTH = 10;
-            }
+        if (strTemp == "none")
+            FETCH_DEPTH = 0;
+        else if (strTemp == "full")
+            FETCH_DEPTH = 1000000000u;
+        else
+            FETCH_DEPTH = beast::lexicalCastThrow <std::uint32_t> (strTemp);
 
-            if (getSingleSection (secConfig, SECTION_PATH_SEARCH_OLD, strTemp))
-                PATH_SEARCH_OLD     = beast::lexicalCastThrow <int> (strTemp);
-            if (getSingleSection (secConfig, SECTION_PATH_SEARCH, strTemp))
-                PATH_SEARCH         = beast::lexicalCastThrow <int> (strTemp);
-            if (getSingleSection (secConfig, SECTION_PATH_SEARCH_FAST, strTemp))
-                PATH_SEARCH_FAST    = beast::lexicalCastThrow <int> (strTemp);
-            if (getSingleSection (secConfig, SECTION_PATH_SEARCH_MAX, strTemp))
-                PATH_SEARCH_MAX     = beast::lexicalCastThrow <int> (strTemp);
-
-            if (getSingleSection (secConfig, SECTION_ACCOUNT_PROBE_MAX, strTemp))
-                ACCOUNT_PROBE_MAX   = beast::lexicalCastThrow <int> (strTemp);
-
-            (void) getSingleSection (secConfig, SECTION_SMS_FROM, SMS_FROM);
-            (void) getSingleSection (secConfig, SECTION_SMS_KEY, SMS_KEY);
-            (void) getSingleSection (secConfig, SECTION_SMS_SECRET, SMS_SECRET);
-            (void) getSingleSection (secConfig, SECTION_SMS_TO, SMS_TO);
-            (void) getSingleSection (secConfig, SECTION_SMS_URL, SMS_URL);
-
-            if (getSingleSection (secConfig, SECTION_VALIDATORS_FILE, strTemp))
-            {
-                VALIDATORS_FILE     = strTemp;
-            }
+        if (FETCH_DEPTH < 10)
+            FETCH_DEPTH = 10;
+    }
 
-            if (getSingleSection (secConfig, SECTION_DEBUG_LOGFILE, strTemp))
-                DEBUG_LOGFILE       = strTemp;
-        }
+    if (getSingleSection (secConfig, SECTION_PATH_SEARCH_OLD, strTemp))
+        PATH_SEARCH_OLD     = beast::lexicalCastThrow <int> (strTemp);
+    if (getSingleSection (secConfig, SECTION_PATH_SEARCH, strTemp))
+        PATH_SEARCH         = beast::lexicalCastThrow <int> (strTemp);
+    if (getSingleSection (secConfig, SECTION_PATH_SEARCH_FAST, strTemp))
+        PATH_SEARCH_FAST    = beast::lexicalCastThrow <int> (strTemp);
+    if (getSingleSection (secConfig, SECTION_PATH_SEARCH_MAX, strTemp))
+        PATH_SEARCH_MAX     = beast::lexicalCastThrow <int> (strTemp);
+
+    if (getSingleSection (secConfig, SECTION_ACCOUNT_PROBE_MAX, strTemp))
+        ACCOUNT_PROBE_MAX   = beast::lexicalCastThrow <int> (strTemp);
+
+    (void) getSingleSection (secConfig, SECTION_SMS_FROM, SMS_FROM);
```

Comment:
We don't generally cast away return values we ignore - we just ignore them.  :-)  Quibble again.